wallet-next/src/storage.rs

```diff
@@ -1,26 +1,78 @@
+use std::path::PathBuf;
+
+use anyhow::anyhow;
 use penumbra_chain::params::ChainParams;
 use penumbra_crypto::{
     merkle::{NoteCommitmentTree, Tree},
     note::Commitment,
     FieldExt,
 };
 use penumbra_proto::{crypto::FullViewingKey, Message, Protobuf};
-use sqlx::{query, Pool, Sqlite};
+use sqlx::{migrate::MigrateDatabase, query, Pool, Sqlite};
 
 use crate::sync::ScanResult;
 
+const MAX_MERKLE_CHECKPOINTS_CLIENT: usize = 10;
+
 #[derive(Clone)]
 pub struct Storage {
     pool: Pool<Sqlite>,
 }
 
 impl Storage {
-    pub fn new(pool: Pool<Sqlite>) -> Self {
-        Self { pool }
+    pub async fn load(storage_path: String) -> anyhow::Result<Self> {
+        Ok(Self {
+            pool: Pool::<Sqlite>::connect(&storage_path).await?,
+        })
     }
 
-    pub async fn migrate(self: &Storage) -> anyhow::Result<()> {
-        sqlx::migrate!().run(&self.pool).await.map_err(Into::into)
+    pub async fn initialize(
+        storage_path: String,
+        fvk: FullViewingKey,
+        params: ChainParams,
+    ) -> anyhow::Result<Self> {
+        //   Check that the file at the given path does not exist;
+        if !PathBuf::from(&storage_path).exists() {
```

Comment:
The eventual error ends up fairly distant at the bottom of the function; I think it'd be cleaner and avoid rightward drift if the conditional were reversed, to do if (exists) { return Err; }, so the error can be local to the conditional and the rest of the happy path can be unindented